consensus/consensus.go

```diff
@@ -89,9 +86,11 @@ type Consensus struct {
 	ShardID uint32
 	// IgnoreViewIDCheck determines whether to ignore viewID check
 	IgnoreViewIDCheck *abool.AtomicBool
-	// global consensus mutex
-	// TODO(optimization): Avoid mutex and use more efficient locking primitives
-	mutex sync.Mutex
+	// consensus mutex
+	prepareMutex sync.RWMutex
+	commitMutex  sync.RWMutex
+	// mutex for view change
+	vcLock sync.Mutex
```

Comment:
vcLock is not exclusive with prepareMutex & commitMutex, but somehow they are actually protecting the same data structure. So this will cause a race condition.

Reply:
yea, you are right, the concurrency between view change and consensus are not strictly protected. But it's semi-protected by the ViewChanging mode which will let the node only process one type of messages (either consensus or view change) in specific mode. So far, the view change and consensus are working fine without concurrency issue. So this PR will not address this separate issue here.